README.md

```diff
@@ -190,3 +191,47 @@ optional arguments:
   -h, --help        show this help message and exit
   --an-integer INT
 ```
+
+## Nested Dataclasses
+
+Dataclasses may be nested; the type of a dataclass field may be another dataclass type:
+
+```python
+from dataclasses import dataclass, field
+
+@dataclass
+class Config:
+  field_a: int
+  field_b: str = "abc"
+
+
+@dataclass
+class Base:
+  config: Config
+  verbose: bool = False
+```
+
+Argument names of fields of the nested dataclass are prefixed with the field name of the nested dataclass in the base
+dataclass. Calling `pydargs.parse(Base, ["-h"])` will result in somthing like:
+
+```text
+usage: your_program.py [-h] --config-field-a CONFIG_FIELD_A
+                            [--config-field-b CONFIG_FIELD_B]
+                            [--verbose VERBOSE]
+
+options:
+  -h, --help            show this help message and exit
+  --verbose VERBOSE     (default: False)
+
+config:
+  --config-field-a CONFIG_FIELD_A
+  --config-field-b CONFIG_FIELD_B
+                        (default: abc)
+
+```
+
+Please be aware of the following:
+- The default (factory) of fields with a dataclass type is ignored by pydargs, which may yield unexpected results.
+  If you must add a default, do `field_name: NestedClass = field(default_factory: NestedClass)`.
```

Comment:
```suggestion
- The default (factory) of fields with a dataclass type is ignored by pydargs, which may yield unexpected results. 
  I.e. in the provided default for `a` in `field_name: NestedClass = field(default_factory: NestedClass(a=1))` is ignored. 
  If you want to set a defaults for the NestedClass, you must do this in the class definition of `NestedClass`
  and set the default for the nested class as `field_name: NestedClass = field(default_factory: NestedClass)`.
```

Reply:
This is a little more explicit, it took me some time to figure out what you meant

Reply:
Is this last line also correct? or is the default factory not required? if not, why would a default_factory be needed? For the ordering of the properties on the class? i.e. properties without defaults after properties with defaults?

Reply:
Your suggestion isn't entirely correct. The note isn't about arguments to a default constructor, but about the entire default and/or default_factory. Whatever is provided as `default` or `default_factory`, is completely ignored.

Would this be better?

```suggestion
- The default (factory) of fields with a dataclass type is ignored by pydargs, which may yield unexpected results.
  E.g., in the example above, `config: Config = field(default_factory=lambda: Config(field_b="def"))` will not result in a default of "def" for field_b when parsed by pydargs.
  If you must add a default, for example for instantiating your dataclass elsewhere, do `config: Config = field(default_factory=Config)`.
```

Reply:
> If you want to set a defaults for the NestedClass, you must do this in the class definition of `NestedClass`
This part of my suggestion is still relevant though, right? 

> If you must add a default, for example for instantiating your dataclass elsewhere, do `config: Config = field(default_factory=Config)`.

Is this strictly required when instantiation the object manually? 

If so, I think we can rephrase it as:
> To support instantiating the dataclass directly, you should do `config: Config = field(default_factory=Config)`.

Reply:
> This part of my suggestion is still relevant though, right?

True; how about this?

```suggestion
- The default (factory) of fields with a dataclass type is ignored by pydargs, which may yield unexpected results.
  E.g., in the example above, `config: Config = field(default_factory=lambda: Config(field_b="def"))` will not result in a default of "def" for field_b when parsed by pydargs. 
  Instead, set `field_b: str = "def"` in the definition of `Config`..
  If you must add a default, for example for instantiating your dataclass elsewhere, do `config: Config = field(default_factory=Config)`, assuming that all fields in `Config` have a default.
```

Having a default for a dataclass-type field is not necessary for instantiating it elsewhere, but it is required if you want to be able to instantiate it without arguments.

Reply:
Yes that looks good! Note that on line `#236` you have two dots. Not sure if intentional